tiatoolbox/wsicore/wsireader.py

```diff
@@ -301,13 +303,138 @@ def find_read_rect_params(self, location, size, resolution, units, precision=3):
+            self, location, size, resolution, units):
+        """Work similar to `find_read_rect_params`.
+
+        This is similar for when location is at requested resolution.
+
+        Args:
+            location (tuple(int)): Location in the requested resolution system.
+            size (tuple(int)): Desired output size in pixels (width,
+                height) tuple and in the requested resolution system.
+            resolution (float): Desired output resolution.
+            units (str): Units of scale, default = "level".
+                Supported units are:
+                - microns per pixel ('mpp')
+                - objective power ('power')
+                - pyramid / resolution level ('level')
+                - pixels per baseline pixel ('baseline')
+            precision (int, optional): Decimal places to use when
+                finding optimal scale. See
+                :func:`find_optimal_level_and_downsample` for more.
+
+        Returns:
+            tuple: Parameters for reading the requested region
+            - :py:obj:`int` - Optimal read level
+            - :py:obj:`tuple` - Scaling to apply after level read to achieve
+              desired output resolution.
+                - :py:obj:`float` - X scale factor
+                - :py:obj:`float` - Y scale factor
+            - :py:obj:`tuple` - Region size in read level coordinates
+                - :py:obj:`int` - Width
+                - :py:obj:`int` - Height
+            - :py:obj:`tuple` - Region location in read level coordinates
+                - :py:obj:`int` - X location
+                - :py:obj:`int` - Y location
+            - :py:obj:`tuple` - Region size in level 0 coordinates
+                - :py:obj:`int` - Width
+                - :py:obj:`int` - Height
+            - :py:obj:`tuple` - Region location level 0 coordinates
+                - :py:obj:`int` - X location
+                - :py:obj:`int` - Y location
+        """
+        (
+            read_level,
+            read_lv_to_requested_fx
+        ) = self._find_optimal_level_and_downsample(
+            resolution, units,
+        )
+        info = self.info
+
+        # Do we need sanity check for input form ?
+        requested_location = np.array(location)
+        requested_size = np.array(size)
+        lv0_to_read_lv_fx = 1 / info.level_downsamples[read_level]
+        lv0_to_requested_fx = lv0_to_read_lv_fx * read_lv_to_requested_fx
+        size_at_lv0 = requested_size / lv0_to_requested_fx
+        location_at_lv0 = requested_location / lv0_to_requested_fx
+        size_at_read_lv = requested_size / read_lv_to_requested_fx
+        location_at_read_lv = requested_location / read_lv_to_requested_fx
+        output = (
+            size_at_read_lv,
+            location_at_read_lv,
+            size_at_lv0,
+            location_at_lv0,
+        )
+        # rounding up
+        output = tuple([(v + 0.5).astype(np.int64) for v in output])
```

Comment:
Elsewhere, when rounding I use `np.round` or python `round` which both do 'bankers' rounding. I think we should use the same rounding function to be consistent (avoid differences in when we round up or down on .5 etc). Unless there is a particular reason why it must round up. If so, then maybe `np.ceil` would be easier to read and not require a comment?